File: pkg/server/pokete/msg/fight/seed.go
package fight

import (
	"github.com/lxgr-linux/pokete/bs_rpc/msg"
)

const SeedType msg.Type = "pokete.fight.seed"

type Seed struct {
	msg.BaseMsg
	Seed uint32 `json:"seed"`
}

func (i Seed) GetType() msg.Type {
	return SeedType
}

func NewSeed(seed uint32) Seed {
	return Seed{msg.BaseMsg{}, seed}
}

File: src/pokete/classes/multiplayer/msg/fight/seed.py
from typing import TypedDict
import pokete.bs_rpc as bs_rpc

SEED_TYPE = "pokete.fight.seed"


class SeedData(TypedDict):
    seed: int


class Seed(bs_rpc.Body):
    def __init__(self, data: SeedData):
        super().__init__(SEED_TYPE, data)
